app/src/main/java/com/example/expenseappmvvm/screens/mainScreen/fragments/expense/ExpenseFragmentViewModel.kt

```diff
@@ -0,0 +1,129 @@
+package com.example.expenseappmvvm.screens.mainScreen.fragments.expense
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.expenseappmvvm.R
+import com.example.expenseappmvvm.data.database.entities.Transaction
+import com.example.expenseappmvvm.data.database.repositories.TransactionRepository
+import com.example.expenseappmvvm.data.rx.AppRxSchedulers
+import com.example.expenseappmvvm.screens.addActionScreen.enums.CategoryEnum
+import com.example.expenseappmvvm.screens.mainScreen.fragments.expense.adapter.models.RecyclerTransaction
+import com.example.expenseappmvvm.utils.Constants
+import com.example.expenseappmvvm.utils.DateUtils
+import com.example.expenseappmvvm.utils.Preferences
+import com.example.expenseappmvvm.utils.disposeBy
+import com.example.expenseappmvvm.utils.resourceUtils.ResourceUtils
+import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
+import java.text.DecimalFormat
+import java.util.*
+
+class ExpenseFragmentViewModel(
+    private val resourceUtils: ResourceUtils,
+    private val sharedPref: Preferences,
+    private val transactionRepository: TransactionRepository,
+    private val rxSchedulers: AppRxSchedulers,
+    private val compositeDisposable: CompositeDisposable
+) : ViewModel() {
+    private var endDay: Long = Constants.DEFAULT_LONG
+    private var userId: Long = Constants.DEFAULT_LONG
+    private val twoDigitsFormatter: DecimalFormat = DecimalFormat(Constants.TWO_DECIMAL_FORMATTER)
+    val startWeek = DateUtils.getEndDate(Calendar.WEEK_OF_MONTH)
+    var totalExpense = MutableLiveData<String>().apply { value = Constants.EMPTY_EXPENSE }
+    val showHide = MutableLiveData<Boolean>().apply { value = false }
+    val orderedList = mutableListOf<RecyclerTransaction>()
+
+    private fun calculateDynamicBalance(
+        orderedList: List<RecyclerTransaction>,
+        userBalance: Double
+    ): List<RecyclerTransaction> {
+        orderedList[0].balance = userBalance
+        orderedList[orderedList.size - 1].balance = orderedList[orderedList.size - 1].amount
+        for (index in 1 until orderedList.size) {
```

Comment:
get used with forEach instead of a simple for, you get the same behaviour but is safer